packages/offix-client/src/config/OffixClientConfig.ts

```diff
@@ -1,99 +1,55 @@
- * Users do not need to pass them for normal initialization of the client.
- * Please refer to documentation for more information about the individual flag and it's side effects.
- *
+ * Class for managing user and default configuration.
+ * Default config is applied on top of user provided configuration
  */
-export interface OffixClientConfig {
-
-  /**
-   * The URL of http server that will be used to initialize default http link
-   * Value is ignored if terminating link is passed
-   */
-  httpUrl?: string;
-
-  /**
-   * [Modifier]
-   * Apollo link that will be passed to created client
-   */
-  terminatingLink?: ApolloLink;
-
-  /**
-   * [Modifier]
-   *
-   * The storage you want your client to use (Uses window.localStorage by default)
-   */
-  storage?: PersistentStore<PersistedData>;
-
-  /**
-   * [Modifier]
-   *
-   * The Apollo InMemoryCache you want your client to use. (Uses a default one if none provided)
-   */
-  cache?: InMemoryCache;
-
-  /**
-   * [Modifier]
-   *
-   * Interface for detecting changes in network status.
-   * See `WebNetworkStatus` and `CordovaNetworkStatus`
-   */
-  networkStatus?: NetworkStatus;
-
-  /**
-   * User provided listener that contains set of methods that can be used to detect
-   * when operations were added to queue
-   */
-  offlineQueueListener?: ApolloOfflineQueueListener;
-
-  /**
-   * [Modifier]
-   *
-   * Interface that defines how object state is progressed
-   * This interface needs to match state provider supplied on server.
-   */
-  conflictProvider?: ObjectState;
-
-  /**
-   * Interface that can be implemented to receive information about the data conflict
-   *
-   * @deprecated see OfflineClient.registerOfflineEventListener
-   */
-  conflictListener?: ConflictListener;
-
-  /**
-   * [Modifier]
-   *
-   * The conflict resolution strategy your client should use. By default it takes client version.
-   */
-  conflictStrategy?: ConflictResolutionStrategy;
-
-  /**
-   * [Modifier]
-   *
-   * Cache updates functions for your mutations
-   * Argument allows to restore optimistic responses on application restarts.
-   */
-  mutationCacheUpdates?: CacheUpdates;
-
-  /**
-   * [Modifier]
-   *
-   * The options to configure how failed offline mutations are retried.
-   *
-   */
-  retryOptions?: RetryLink.Options;
+export class OffixClientConfig implements OffixClientOptions {
+  public httpUrl?: string;
+  public offlineQueueListener?: ApolloOfflineQueueListener;
+  public conflictStrategy: ConflictResolutionStrategy;
+  public conflictProvider = new VersionedState();
+  public networkStatus?: NetworkStatus;
+  public terminatingLink: ApolloLink | undefined;
+  public cacheStorage?: PersistentStore<PersistedData>;
+  public offlineStorage?: PersistentStore<PersistedData>;
+  public conflictListener?: ConflictListener;
+  public mutationCacheUpdates?: CacheUpdates;
+  public cache?: InMemoryCache;
+
+  public retryOptions = {
```

Comment:
[Question] Do we actually still support retries (retry link) after moving to scheduler?

Reply:
We do in OfflineClient using the Apollo RetryLink. The scheduler doesn't have it though. It's something we should definitely be looking at.